logic/get_user.php

```diff
@@ -6,39 +6,224 @@
+        $msg .= getTranslation('team') . ': ' . $GLOBALS['teams'][$row['team']] . CR;
     }
 
     // Add level.
     if ($row['level'] != 0) {
-        $msg .= 'Level: <b>' . $row['level'] . '</b>' . CR;
+        $msg .= getTranslation('level') . ': <b>' . $row['level'] . '</b>' . CR;
     }
 
     return $msg;
 }
 
+/**
+ * Delivers Trainername (if not set) ->  Telegram-@Nick (if not set) -> Telegram-name
+ * @param array $row
+ * @return array $row
+ */
+function check_trainername($row){
+    global $config;
+    if($config->CUSTOM_TRAINERNAME==true){ // if Custom Trainername is enabled by config
+        if(check_for_empty_string($row['trainername'])){ // trainername not set by user
+            // check if Telegram-@Nick is set
+              if(check_for_empty_string($row['nick'])){
+                // leave Telegram-name as it is (Trainername and Telegram-@Nick were not configured by user)
+              }else{
+                // set Telegram-@Nick as Name inside the bot
+                $row['name'] = $row['nick'];
+              }
+        }else{
+            // Trainername is configured by User
+            $row['name'] = $row['trainername'];
+        }
+    }else{ // Custom Trainername is disabled by config
+      // check if Telegram-@Nick is set
+      if(check_for_empty_string($row['nick'])){
+        // do nothing -> leave Telegram-name
+      }else{
+        // set Telegram-@Nick as Name inside the bot
+        $row['name'] = $row['nick'];
+      }
+    }
+
+    return $row;
+}
+
+/**
+ * Checks if String is empty
+ * @param String $string
+ * @return boolean | true = empty | false = not empty
+ */
+function check_for_empty_string($string){
+  if($string == "" || is_null($string) || empty($string)){
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Changes the trainername
+ * @param array $update
+ * @param array $answer
+ */
+function change_trainername($update, $answer){
+    // get UserID from Message
+    $userid = $update['message']['from']['id'];
+
+	if($answer['user_id'] == $userid) // Check if Answer is for the right User
+	{
+		$returnValue = preg_match('/^[A-Za-z0-9]{0,15}$/', $update['message']['text']);
+        // Only numbers and alphabetic character allowed
+		if($returnValue){
+            $trainername = $update['message']['text'];
+            // Store new Gamer-Name to DB
+			my_query(
+                "
+                UPDATE users
+                SET trainername_time =  NULL,
+                    trainername =   '{$trainername}'
+                WHERE user_id =     {$userid}
+                "
+            );
+            // confirm Name-Change
+			sendMessage($userid, getTranslation('trainername_success').' <b>'.$trainername.'</b>');
+		}else{
+            // Trainer Name got unallowed Chars -> Error-Message
+			sendMessage($userid, getTranslation('trainername_fail'));
+            // Set trainername_time to 'still waiting for Name-Change'
```

Comment:
This is the time, the user got to change the name. Because the bots is waiting for a message and there is no input-window. The `function evaluate_priv_message($update)` will check the database if the time interval is active or expired.